timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureLockWatchDiagnosticsResource.java

```diff
@@ -0,0 +1,80 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.atlasdb.futures.AtlasFutures;
+import com.palantir.atlasdb.http.RedirectRetryTargeter;
+import com.palantir.atlasdb.timelock.lock.watch.ConjureLockWatchDiagnosticsService;
+import com.palantir.atlasdb.timelock.lock.watch.ConjureLockWatchDiagnosticsServiceEndpoints;
+import com.palantir.atlasdb.timelock.lock.watch.UndertowConjureLockWatchDiagnosticsService;
+import com.palantir.conjure.java.undertow.lib.RequestContext;
+import com.palantir.conjure.java.undertow.lib.UndertowService;
+import com.palantir.tokens.auth.AuthHeader;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+public final class ConjureLockWatchDiagnosticsResource implements UndertowConjureLockWatchDiagnosticsService {
+    private final ConjureResourceExceptionHandler exceptionHandler;
+    private final BiFunction<String, Optional<String>, AsyncTimelockService> timelockServices;
+
+    private ConjureLockWatchDiagnosticsResource(
+            RedirectRetryTargeter redirectRetryTargeter,
+            BiFunction<String, Optional<String>, AsyncTimelockService> timelockServices) {
+        this.exceptionHandler = new ConjureResourceExceptionHandler(redirectRetryTargeter);
+        this.timelockServices = timelockServices;
+    }
+
+    public static UndertowService undertow(
+            RedirectRetryTargeter redirectRetryTargeter,
+            BiFunction<String, Optional<String>, AsyncTimelockService> timelockServices) {
+        return ConjureLockWatchDiagnosticsServiceEndpoints.of(
+                new ConjureLockWatchDiagnosticsResource(redirectRetryTargeter, timelockServices));
+    }
+
+    public static ConjureLockWatchDiagnosticsService jersey(
+            RedirectRetryTargeter redirectRetryTargeter,
+            BiFunction<String, Optional<String>, AsyncTimelockService> timelockServices) {
+        return new JerseyAdapter(new ConjureLockWatchDiagnosticsResource(redirectRetryTargeter, timelockServices));
+    }
+
+    @Override
+    public ListenableFuture<Void> dumpState(AuthHeader authHeader, String namespace, RequestContext requestContext) {
+        return exceptionHandler.handleExceptions(() -> dumpStateSync(namespace, requestContext));
+    }
+
+    private ListenableFuture<Void> dumpStateSync(String namespace, RequestContext context) {
```

Comment:
A log line here with the namespace might help the cross-examination of logs

Reply:
I will add a log line here

For the `sync` method, I was mirroring https://github.com/palantir/atlasdb/blob/d135d021cb8eb59581392519dae362e576a7336a/timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureLockWatchingResource.java#L64. Agreed on the weirdness, I'll try and change it!

Reply:
I think we do this to use the Exception Handler which takes a supplier of future